TFG-Labs/custom-shipping-step-by-items
# [WEB-438] - Improvement/web 436/debounce autocomplete result

### What problem is this solving?
Decreases Autocomplete requests by ensuring input processing is delayed with debounce and limit autocomplete functionality to run once input values/ length is larger than 2.

#### How it works

#### How to test it?
when you click on edit and add address in checkout, when you type in an address google address autocomplete should not run until input length is greater than 2, and should be removed when input length is smaller than 3.
[Workspace](https://web438debounce--thefoschini.myvtex.com/checkout/#/payment) 

### Screenshots/Video example usage:
- Desktop
- Tablet
- Movil

<img width="1411" alt="Screenshot 2024-07-16 at 11 07 07" src="https://github.com/user-attachments/assets/dac597fd-9e9c-4c88-874c-e37b1c74e72e">
<img width="1411" alt="Screenshot 2024-07-16 at 11 07 20" src="https://github.com/user-attachments/assets/036a8b4c-330f-4a1e-8014-fa88ff199099">


https://github.com/user-attachments/assets/0d395af4-e3a8-4d87-a939-5215c6674242



### Related to / Depends on

#### How does this PR make you feel? :link:
![]()
